src/pick.c

```diff
@@ -729,3 +792,15 @@ free_choices(void)
 	free(choices.v);
 	free(input.string);
 }
+
+int
+isu8cont(unsigned char c)
+{
+	return (c & (0x80 | 0x40)) == 0x80;
+}
+
+int
+isu8start(unsigned char c)
```

Comment:
On Fri, Apr 08, 2016 at 01:41:23AM -0700, Calle Erlandsson wrote:
> > @@ -729,3 +792,15 @@ free_choices(void)
> >  	free(choices.v);
> >  	free(input.string);
> >  }
> > +
> > +int
> > +isu8cont(unsigned char c)
> > +{
> > +	return (c & (0x80 | 0x40)) == 0x80;
> > +}
> > +
> > +int
> > +isu8start(unsigned char c)
> 
> Does this function return whether or not the byte is the continuation of an utf-8 character?

It returns 1 if the given byte marks the start of a UTF-8 character: the
two MSBs are both set to 1 as opposed of a continuation byte where the
two MSBs are equal to `10'.